app/src/main/java/org/mozilla/tv/firefox/architecture/KillswitchLinearLayout.kt

```diff
@@ -71,21 +69,38 @@ class KillswitchLinearLayout : LinearLayout {
             return super.setVisibility(View.GONE)
         }
 
-        val allAllowed = allowedInLocales == KillswitchLocales.All
+        if (allowedInLocales == KillswitchLocales.All) {
+            return super.setVisibility(visibility)
+        }
+
+
         val allowedLocales = (allowedInLocales as? KillswitchLocales.ActiveIn)?.locales
         val currentLocale = localeManager.getCurrentLocale(context)
 
-        val currentLocaleIsAllowed = (allowedLocales != null &&
-            allowedLocales.any { allowed ->
-                allowed.language == currentLocale.language &&
-                    (allowed.country.isEmpty() ||
-                        allowed.country == currentLocale.country)
-            })
+        val isCurrentLocaleAllowed = currentLocale.languageAndMaybeCountryMatch(allowedLocales)
 
-        if (!allAllowed && !currentLocaleIsAllowed) {
+        if (!isCurrentLocaleAllowed) {
             return super.setVisibility(View.GONE)
         }
 
         super.setVisibility(visibility)
     }
 }
+
+/**
+ * Returns true if the language and country of the current locale matches any
+ * one of the allowed locales.
+ *
+ * An allowed locale with no country set (represented by an empty string, see
```

Comment:
Can we document this behavior on a non-private method for consumers of the layout? Maybe `setRequirements`?